stdlib/Random/src/normal.jl

```diff
@@ -90,6 +90,15 @@ randn(rng::AbstractRNG, ::Type{Complex{T}}) where {T<:AbstractFloat} =
     Complex{T}(SQRT_HALF * randn(rng, T), SQRT_HALF * randn(rng, T))
 
 
+### fallback randn for float types defining rand:
+function randn(rng::AbstractRNG, ::Type{T}) where {T<:AbstractFloat}
+    # Marsaglia polar variant of Box–Muller transform:
+    while true
+        x, y = 2rand(rng, T)-1, 2rand(rng, T)-1
+        0 < (s = x^2 + y^2) < 1 && return x * sqrt(-2log(s)/s)
```

Comment:
Maybe check also for `!iszero(s)` ? (Although it seems incredibly unlikely...)

Reply:
It's already checking `0 < s < 1`, so `iszero` would be redundant.

I guess I could check `!iszero(s)` instead of `0 < s`, in the hope that the former might be slightly more efficient?   But the performance difference would surely be negligible compared to the cost of the `rand` and `sqrt` and `log` calls, and it would make the code messier.

Reply:
Ah sorry, I read too fast and missed the `0 < ` part.